src/hal/src/adapter.rs

```diff
@@ -116,11 +116,12 @@ pub trait PhysicalDevice<B: Backend>: fmt::Debug + Any + Send + Sync {
     fn memory_properties(&self) -> MemoryProperties;
 
     /// Returns the features of this `PhysicalDevice`. This usually depends on the graphics API being
-    /// used.
+    /// used, as well as the actual platform underneath.
     fn features(&self) -> Features;
 
-    /// Returns the performance hints of this `PhysicalDevice`.
-    fn hints(&self) -> Hints;
+    /// Returns the capabilities of this `PhysicalDevice`. Similarly to `Features`, they
```

Comment:
I wonder if it would make sense to collapse `Features` and `Capabilities` together so there's only one place to look. e.g. a user could still request these dynamic states if we exposed it as a feature instead, and we'd just fail in that case

Reply:
The idea I wanted to realize here is a structure with stuff that does *not* need to be enabled. Currently, it's either in Features, which have to be enabled explicity, or Limits, which are 99% just numbers. So putting the dynamic state in either was not a good option. But we had those `Hints`, so I replaced them by a structure that is more scalable. Technically, we could move `Limits` in there as well. Do you think it's worth doing? I don't have a strong feeling here.

Reply:
Right, I don’t have a strong preference either. I was just thinking it’s a little unfortunate that the users have to look in multiple places to check which features/capabilities they can use, but maybe that’s fine